test/notebooks/test_notebooks.py

```diff
@@ -0,0 +1,134 @@
+LEARNING_NOTEBOOKS = [
+    EXAMPLES_DIR / "learning" / "learning_gridworld.ipynb",
+]
+
+# Inference notebooks
+INFERENCE_NOTEBOOKS = [
+    EXAMPLES_DIR / "inference_and_learning" / "inference_methods_comparison.ipynb",
+]
+
+# Inductive inference notebooks
+INDUCTIVE_NOTEBOOKS = [
+    EXAMPLES_DIR / "inductive_inference" / "inductive_inference_example.ipynb",
+    EXAMPLES_DIR / "inductive_inference" / "inductive_inference_gridworld.ipynb",
+]
+
+# Advanced feature notebooks
+ADVANCED_NOTEBOOKS = [
+    EXAMPLES_DIR / "advanced" / "complex_action_dependency.ipynb",
+]
+
+# Model fitting notebooks
+MODEL_FITTING_NOTEBOOKS = [
+    EXAMPLES_DIR / "model_fitting" / "fitting_with_pybefit.ipynb",
+]
+
+# Sparse computation notebooks
+SPARSE_NOTEBOOKS = [
+    EXAMPLES_DIR / "sparse" / "sparse_benchmark.ipynb",
+]
+
+# Experimental notebooks (may be unstable)
+EXPERIMENTAL_NOTEBOOKS = [
+    EXAMPLES_DIR / "experimental" / "sophisticated_inference" / "si_graph_world.ipynb",
+    EXAMPLES_DIR / "experimental" / "sophisticated_inference" / "si_generalized_tmaze.ipynb",
+    EXAMPLES_DIR / "experimental" / "sophisticated_inference" / "mcts_graph_world.ipynb",
+    EXAMPLES_DIR / "experimental" / "sophisticated_inference" / "mcts_generalized_tmaze.ipynb",
+]
+
+
+@pytest.mark.notebook
+@pytest.mark.fast
+@pytest.mark.parametrize("notebook", FAST_NOTEBOOKS)
+def test_fast_notebooks(notebook):
+    """Test fast-running notebooks suitable for every commit."""
+    assert notebook.exists(), f"Notebook not found: {notebook}"
+
+
+@pytest.mark.notebook
+@pytest.mark.slow
+@pytest.mark.parametrize("notebook", ENV_NOTEBOOKS)
+def test_env_notebooks(notebook):
+    """Test environment demonstration notebooks."""
+    assert notebook.exists(), f"Notebook not found: {notebook}"
+
+
+@pytest.mark.notebook
+@pytest.mark.slow
+@pytest.mark.parametrize("notebook", LEARNING_NOTEBOOKS)
+def test_learning_notebooks(notebook):
+    """Test learning-related notebooks."""
+    assert notebook.exists(), f"Notebook not found: {notebook}"
+
+
+@pytest.mark.notebook
+@pytest.mark.slow
+@pytest.mark.parametrize("notebook", INFERENCE_NOTEBOOKS)
+def test_inference_notebooks(notebook):
+    """Test inference method notebooks."""
+    assert notebook.exists(), f"Notebook not found: {notebook}"
+
+
+@pytest.mark.notebook
+@pytest.mark.slow
+@pytest.mark.parametrize("notebook", INDUCTIVE_NOTEBOOKS)
+def test_inductive_notebooks(notebook):
+    """Test inductive inference notebooks."""
+    assert notebook.exists(), f"Notebook not found: {notebook}"
+
+
+@pytest.mark.notebook
+@pytest.mark.slow
+@pytest.mark.parametrize("notebook", ADVANCED_NOTEBOOKS)
+def test_advanced_notebooks(notebook):
+    """Test advanced feature notebooks."""
+    assert notebook.exists(), f"Notebook not found: {notebook}"
+
+
+@pytest.mark.notebook
+@pytest.mark.slow
+@pytest.mark.skip("model fitting notebook is currently being migrated, see https://github.com/infer-actively/pymdp/issues/287")
```

Comment:
I think this pytest.mark.skip can be removed now, because the notebook is currently running (but the pytest.mark.slow is certainly warranted)